src/test/java/org/folio/rest/impl/PieceApiTest.java

```diff
@@ -47,4 +51,62 @@ public void testPostPiece() {
     verifyPostResponse(PIECES_ENDPOINT, JsonObject.mapFrom(postPieceRq).encode(), prepareHeaders(EXIST_CONFIG_X_OKAPI_TENANT_LIMIT_10,
       new Header(X_ECHO_STATUS, String.valueOf(status500))), APPLICATION_JSON, status500);
   }
+  
+  @Test
+  public void testPutPiecesByIdTest() throws Exception {
+    logger.info("=== Test update piece by id - valid Id 204 ===");
+    
+    String reqData = getMockData(PIECE_RECORDS_MOCK_DATA_PATH + "pieceRecord.json");
+
+    verifyPut(String.format(PIECES_ID_PATH, VALID_UUID), reqData, "", 204);
+  }
+
+  @Test
+  public void testPutPiecesByNonExistentId() throws Exception {
+    logger.info("=== Test update piece by id - Id does not exists 404 ===");
+    
+    Piece reqData = pieceJsonReqData.mapTo(Piece.class);
+    reqData.setId(ID_DOES_NOT_EXIST);
+    String jsonBody = JsonObject.mapFrom(reqData)
+      .encode();
+
+    verifyPut(String.format(PIECES_ID_PATH, ID_DOES_NOT_EXIST), jsonBody, APPLICATION_JSON, 404);
+  }
+
+  @Test
+  public void testPutPiecesWithError() throws Exception {
+    logger.info("=== Test update piece by id - internal error from storage 500 ===");
+    
+    Piece reqData = pieceJsonReqData.mapTo(Piece.class);
+    reqData.setId(ID_FOR_INTERNAL_SERVER_ERROR);
+    String jsonBody = JsonObject.mapFrom(reqData)
+      .encode();
+
+    verifyPut(String.format(PIECES_ID_PATH, ID_FOR_INTERNAL_SERVER_ERROR), jsonBody, APPLICATION_JSON, 500);
```

Comment:
The raml was having a return type of "text/plain" , how did the expected type "application/json" work?

Reply:
ok, NVM, we are actually returning an application/json from code irrespective of the type specified in the raml [buildErrorResponse](https://github.com/folio-org/mod-orders/blob/24f9b20dab3cd1382e9c58dc195645f4bded2990/src/main/java/org/folio/rest/impl/AbstractHelper.java#L203)